lib/services/resource.go

```diff
@@ -203,6 +203,7 @@ const (
 	VerbRotate = "rotate"
 )
 
+// CollectOptions collects all options from functional arg and returns config
 func CollectOptions(opts []MarshalOption) (*MarshalConfig, error) {
```

Comment:
I'm a little unclear about why we have both an exported version of CollectOptions and the unexported collectOptions which are identical functions...